tensorboard/components_polymer3/tf_dashboard_common/data-loader-behavior.ts

```diff
@@ -192,62 +192,51 @@ export function DataLoaderBehavior<Item, Data>(
           if (result.cancelled) {
             return;
           }
-          // Read-only property have a special setter.
           this.dataLoading = true;
-          // Promises return cacheKeys of the data that were fetched.
-          const promises = this.dataToLoad
-            .filter((datum) => {
-              const cacheKey = this.getDataLoadName(datum);
-              return !this._dataLoadState.has(cacheKey);
-            })
-            .map((datum) => {
-              const cacheKey = this.getDataLoadName(datum);
-              this._dataLoadState.set(cacheKey, LoadState.LOADING);
-              return this.requestData(datum).then(
-                this._canceller.cancellable((result) => {
-                  // It was resetted. Do not notify of the response.
-                  if (!result.cancelled) {
-                    this._dataLoadState.set(cacheKey, LoadState.LOADED);
-                    this.loadDataCallback(this, datum, result.value as any);
-                  }
-                  return cacheKey;
-                })
-              );
-            });
-          return Promise.all(promises)
-            .then(
-              this._canceller.cancellable((result) => {
-                // It was resetted. Do not notify of the data load.
-                if (!result.cancelled) {
-                  const keysFetched = result.value as any;
-                  const fetched = new Set(keysFetched);
-                  const shouldNotify = this.dataToLoad.some((datum) =>
-                    fetched.has(this.getDataLoadName(datum))
-                  );
-                  if (shouldNotify) {
-                    this.onLoadFinish();
-                  }
-                }
-                const isDataFetchPending = Array.from(
-                  this._dataLoadState.values()
-                ).some((loadState) => loadState === LoadState.LOADING);
-                if (!isDataFetchPending) {
-                  // Read-only property have a special setter.
-                  this.dataLoading = false;
-                }
-              }),
-              // TODO(stephanwlee): remove me when we can use
-              // Promise.prototype.finally instead
-              () => {}
-            )
-            .then(
-              this._canceller.cancellable(({cancelled}) => {
-                if (cancelled) {
-                  return;
+          const dirtyItems = this.dataToLoad.filter((datum) => {
```

Comment:
I do not remember your plan so I will ask here: what is our plan with the maximum number of runs we batch?

On the `reload`, we expunge the cache (maybe we should only expunge already loaded so we do not double fetch) and invoke `_loadData`. If user has thousands of runs, one batch request can ask for thousand entries which I don't think will scale nicely (if you want some anecdote, please ask offline). Do we plan on breaking the request down into chunks even in this model?

Reply:
Yeah, fair question. The data loader behavior shouldn’t impose a max
batch size, but I was considering adding a batch size of ~100 in the
`requestData` implementation that calls `/scalars_multirun`. This is
just based off some napkin math: per time series, 1000 points, with
JSON-encoded wall time (~18 bytes), step (~5 bytes), value (~18 bytes),
and overhead (~8 bytes), for about 48 bytes per datum and thus 4.8 KB
per time series. A response size of 500 KB seems entirely reasonable,
and capping it somewhere seems wise to avoid unbounded server load.

> Do we plan on breaking the request down into chunks even in this
> model?

I think that it’s fine to call `requestData` with all the items and let
`requestData` decide if/how to break that up into smaller requests.

Reply:
That plan sounds good. Thanks for sharing your plans.